### ISSUE TITLE
Add Artist Credit to Song, Name sort to Artist

### ISSUE THREAD

**Issue by SenorSmartyPants** [ampache/ampache#1411]
# Environment

 * Ampache version: 3.8.3-develop
 * Web server + version: lighttpd
 * Server operating system: debian
 * Client operating system:

# Description

I would like a field to store Artist Credit in the song. This will be useful for multiple artist support. Here are some example credits

Prince and The Revolution
Prince & The Revolution
Eminem feat. Obie Trice
The Magician Feat. Years & Years

This would be an exact copy of what is in the Artist ID3 tag. Why would we want this when that is sorted in the artist table. First, Artist Credit would not be altered from the ID3 at all (no removal of featuring, etc). When you look at musicbrainz [http://musicbrainz.org/recording/715138fa-c498-4fce-883d-f5e1f6e90797] you see they display the full credit and then the individual artists inside the credit are clickable. We could do that as well once multiple artists are implemented. But even before that, just preserving this information is useful while also not messing up removing duplicate artists.

This would address some of #1402 where my fix is destroying any artist credit detail.

Also useful would be to alter artist so instead of name and prefix fields, we have name and nameSort. We can just dump in the info that MB provides and figure it out when we don't have it.

**Comment by Phyks** [comment]
Hmm, I am not sure to fully understand the possible use case here. :/

**Comment by Rouzax** [comment]
If it would bring Ampache closer to the way MusicBrainz stores the info than I'm for it.

In the latest version of Picard songs are tagged with Artist and Artists.
![image](https://cloud.githubusercontent.com/assets/4103090/19678170/ed8c2b86-9a9c-11e6-99a0-ef4a44cf6021.png)
* Artist is display name that should be displayed to the user
* Artists are all the contributing artists in a multi list format, those should be used to store in the database.
Artists also has the MBID's in there as described in #1402

**Comment by Phyks** [comment]
That is a lot of work, since the DB scheme does not match exactly the id3 tags.

Typically, there are three tables, for artists, albums and songs, and one to one relationships between them. We need to move to a one to many relationship and that is a huge rework of the database.

**Comment by SenorSmartyPants** [comment]
Yes, this would be to start making the database more like Musicbrainz data and properly support multiple artists per track.

Yes, there would be significant work in updating the database (and model classes, etc). But it is a conceptually a relatively straightforward change.

**Comment by Phyks** [comment]
Sure. Problem is that at this time, we are very few to maintain Ampache. So in the near future, we will try to fix bugs, and "enhancements" tagged issues are not the most likely to be handled.

Also, for a couple of months now, we plan a major rewrite of Ampache codebase. There is a huge technical debt in the actual codebase, which makes it quite unstable sometimes, difficult to maintain and difficult for new comers to maintain. Plan is to move the codebase to Laravel framework, use a minimal core codebase and convert every advanced feature as a plugin, maintained separately, so that the whole Ampache code won't break if say Kodi streaming is outdated. This issue would fit perfectly in the refactor frame :)

**Comment by SenorSmartyPants** [comment]
A refactor would definitely be the place to do this. Is there a document on the design of the rewrite? I might be able to contribute time to working on tasks when the rewrite starts.

**Comment by Phyks** [comment]
Not yet, sadly :/
